### ISSUE TITLE
type: 'decay' appears to be broken

### ISSUE THREAD

**Issue by 0xAsimetriq** [nandorojo/moti#321]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Do you want this issue prioritized?

- [ ] Yes, I have sponsored
- [X] Not urgent

### Current Behavior

React native reanimated library exposes `withDecay` helper documented here https://docs.swmansion.com/react-native-reanimated/docs/animations/withDecay it appears from the source that moti also supports this by setting transition to `type: 'decay'` however, doing so breaks the app. with following error:

```
Unhandled Runtime Error
TypeError: animation.callback is not a function
```

### Expected Behavior

Should not break and allow to use decay animations.

### Steps To Reproduce

1. Visit hello world example at https://moti.fyi/hello-world
2. Change transition type from timing to decay in the example and save
3. Test with other values like timing and spring, seems to work ok

### Versions

```markdown
- Moti: 0.27.0
- Reanimated: 3.5.4
- React Native: 0.72.6
```

### Screenshots

https://github.com/nandorojo/moti/assets/3154053/8ec72422-fe31-4fea-88d3-04df5114d13c

### Reproduction

https://stackblitz.com/edit/nextjs-9sshqj?file=pages%2Findex.tsx

**Comment by 0xAsimetriq** [comment]
Related discussions that can be closed if this is an actual bug in favor of this issue.

https://github.com/nandorojo/moti/discussions/319
https://github.com/nandorojo/moti/discussions/313

**Comment by nandorojo** [comment]
Is it web only? Can you confirm it works with normal reanimated but just not moti?

**Comment by 0xAsimetriq** [comment]
Bug initially popped up for me on ios, validated it with web playground. I'll whip up a test with vanilla reanimated in a moment to confirm 👍

**Comment by nandorojo** [comment]
Cool thanks

**Comment by 0xAsimetriq** [comment]
Alright, reproduction for this actually needs to be hooked to gesture (that's how `withDecay`) works essentially. With reanimated, this behaves as expected

```ts
const pan = Gesture.Pan()
    .onChange(({ translationY }) => {
      translate.value = translationY;
    })
    .onFinalize(({ velocityY }) => {
        translate.value = withDecay({
           velocity: velocityY
        });
    })
```

But this (what I believe is equivalent with moti) throws an error when it hits animation with decay
```ts
const pan = Gesture.Pan()
    .onChange(({ translationY }) => {
      animationState.animateTo({ translateY: translationY, transition: { type: 'no-animation' } })
    })
    .onFinalize(({ velocityY }) => {
        animationState.animateTo({ translateY: 0, transition: { type: 'decay', velocity: velocityY } })
    })
```

**Comment by nandorojo** [comment]
How does reanimated perform if you add these values:

```tsx

      velocity: 2,
      deceleration: 2,
```

**Comment by 0xAsimetriq** [comment]
It seems to bug out / shoot everything off to the distance 😅

I copied and set up example from reanimated in this expo snack
https://snack.expo.dev/@asimetriq/juicy-indigo-raspberries

Set values back to normal, but you can set velocity and decay to 2 on line ~30 to see the effect

**Comment by nandorojo** [comment]
Can you try `moti@0.26.1`?

**Comment by nandorojo** [comment]
I'm wondering if `withDecay` is broken on Web altogether from reanimated

**Comment by 0xAsimetriq** [comment]
Getting same error on 0.26.1 as well (testing on ios with fresh node_modules install and ios project rebuild)

**Comment by nandorojo** [comment]
How odd, and it works with plain reanimated?

**Comment by 0xAsimetriq** [comment]
Yes, working example is in the snack I provided above by copying from reanimated's docs. (also tested locally on my own implementation).
